clarify multiple CIDFI-Nonce attributes can appear in a single STUN Indication (#59)
danwing/cidfi: draft-wing-cidfi.md

  |  1. QUIC Initial, transport parameter=CIDFI           |
  +------------------------------------------------------>|
  |  2. STUN Indication, nonce=12345, hmac=e8FEc          |
  +------------------------------------------------------>|
  |                                    |                  |
  |                                    |           3. discarded
  |                                    |                  |
  |                 4. "I saw my nonce, HMAC is valid"    |
  |                                    |                  |
  |  5. Valid STUN Indication processed|                  |
  |<-----------------------------------+                  |
  |                                    |                  |
  |  6. HTTPS: "Map DCID=xyz as high importance"          |
  +----------------------------------->|                  |
  |  7. QUIC Initial, transport parameter=CIDFI           |
  |<------------------------------------------------------+
  |  8. HTTPS: Ok                      |                  |
  |<-----------------------------------+                  |
~~~~~
{: #flow-diag-connect title="Example of Flow to New Server" artwork-align="center"}

> > Note the above message

Reviewer: ```suggestion
> Note that the above message
```

flow shows an initial QUIC handshake for simplicity (steps 1 and 7)

> but because of QUIC connection migration ({{Section 9 of QUIC}}) the

Reviewer: ```suggestion
but, because of QUIC connection migration ({{Section 9 of QUIC}}), the
```

to distinguish each QUIC client's UDP 4-tuple.

To reduce CIDFI setup time the client STUN Indication MAY be sent at
the same time as it establishes connection with the QUIC or DTLS server.

To prevent replay attacks, the Nonce is usable only for authenticating
one UDP 4-tuple.  When the connection is migrated ({{Section 9 of
QUIC}}) the CNE won't apply any CIDFI behavior to
that newly-migrated connection.  The client will have to restart
CIDFI procedures at the beginning ({{attach}}).

After the CIDFI Network Element receives the STUN Indication it
informs the client by sending an HTTP message to the client.  Details TBD.

As the proof of ownership of its UDP 4-tuple is only useful to CIDFI
Network Elements near the client, the client MAY reduce traffic to the

> server by modulating the IP TTL of its STUN Indication messages.

Reviewer: ```suggestion
server by modulating the IPv4 TTL or IPv6 Hop Limit of its STUN Indication messages.
```

> 

Reviewer: The client SHOULD set TTL/Hop Limit to "min-ttl". The client MAY use other values (e.g., explicit configuration, inferred from probe messages).

  HMAC-output = HMAC-SHA256( hmac-secret, nonce | "cidfi" )
~~~~~

When there are multiple CIDFI Network Elements on the network,

> multiple CIDFI-NONE attributes are sent in a single STUN Indication

Reviewer: ```suggestion
multiple CIDFI-NONCE attributes are sent in a single STUN Indication
```

  HMAC-output = HMAC-SHA256( hmac-secret, nonce | "cidfi" )
~~~~~

When there are multiple CIDFI Network Elements on the network,

> multiple CIDFI-NONE attributes are sent in a single STUN Indication

Reviewer 2: Replace "CIDFI-NONE" with "CIDFI-NONCE".
